android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -598,9 +598,16 @@ public void setAdblockRules(WebView view, @Nullable ReadableArray rules) {
   @Override
   protected void addEventEmitters(ThemedReactContext reactContext, WebView view) {
     // Do not register default touch emitter and let WebView implementation handle touches
-    if (((RNCWebView)view).mRNCWebViewClient == null) {
-      view.setWebViewClient(new RNCWebViewClient(reactContext));
+    RNCWebViewClient currentClient = ((RNCWebView)view).mRNCWebViewClient;
+    RNCWebViewClient newClient = new RNCWebViewClient(reactContext);
+
+    if (currentClient != null) {
+      // Client was setup before in onCreateWindow
+      // However it has some override methods, so we have to replace it by a default client
+      // ==> Transfer adblock setting before replacing
+      newClient.cloneAdblockRules(currentClient);
     }
+    view.setWebViewClient(newClient);
```

Comment:
We have to create a new default client because the existing client (created from create window) has some override methods.
Because of the override methods, the redirect behavior doesn't work correctly and cause 2 new window in issue https://github.com/lunascape/phoebe-mobile/issues/2120